fluentgen/src/main/java/com/azure/autorest/fluent/model/arm/ErrorClientModel.java

```diff
@@ -0,0 +1,99 @@
+        .packageName("com.azure.core.management.exception")
+        .properties(Arrays.asList(
+            new ClientModelProperty.Builder()
+                .name("code")
+                .serializedName("code")
+                .description("The error code parsed from the body of the http error response.")
+                .readOnly(true)
+                .wireType(ClassType.STRING)
+                .clientType(ClassType.STRING)
+                .build(),
+            new ClientModelProperty.Builder()
+                .name("message")
+                .serializedName("message")
+                .description("The error message parsed from the body of the http error response.")
+                .readOnly(true)
+                .wireType(ClassType.STRING)
+                .clientType(ClassType.STRING)
+                .build(),
+            new ClientModelProperty.Builder()
+                .name("target")
+                .serializedName("target")
+                .description("The target of the error.")
+                .readOnly(true)
+                .wireType(ClassType.STRING)
+                .clientType(ClassType.STRING)
+                .build(),
+            new ClientModelProperty.Builder()
+                .name("details")
+                .serializedName("details")
+                .description("Details for the error.")
+                .readOnly(true)
+                .wireType(new ListType(FluentType.MANAGEMENT_ERROR))
+                .clientType(new ListType(FluentType.MANAGEMENT_ERROR))
+                .build(),
+            new ClientModelProperty.Builder()
+                .name("additionalInfo")
+                .serializedName("additionalInfo")
+                .description("Additional info for the error.")
+                .readOnly(true)
+                .wireType(new ListType(FluentType.ADDITIONAL_INFO))
+                .clientType(new ListType(FluentType.ADDITIONAL_INFO))
+                .build()
+        )).build();
+
+    private static final ClientModel ADDITIONAL_INFO = new ClientModel.Builder()
+        .name(FluentType.ADDITIONAL_INFO.getName())
+        .packageName("com.azure.core.management.exception")
+        .properties(Arrays.asList(
+            new ClientModelProperty.Builder()
+                .name("type")
+                .serializedName("type")
+                .description("The type of additional info.")
+                .readOnly(true)
+                .wireType(ClassType.STRING)
+                .clientType(ClassType.STRING)
+                .build(),
+            new ClientModelProperty.Builder()
+                .name("info")
+                .serializedName("info")
+                .description("The additional info.")
+                .readOnly(true)
+                .wireType(ClassType.OBJECT)
+                .clientType(ClassType.OBJECT)
+                .build()
+        )).build();
+
+    public static Optional<ClientModel> getErrorClientModel(String modelName) {

```

Comment:
Seeing how this return is used, not a big fan of the `return Optional.empty()` only to check `optional.isPresent()` then `optional.get()` as this is equivalent to `return null` and checking `value != null`.

Reply:
Thanks Alan. 
Seems like I can chain the logic together with `ResourceClientModel.getResourceClientModel` like this:
```java
clientModel = ResourceClientModel.getResourceClientModel(name)
    .or(() -> ErrorClientModel.getErrorClientModel(name))
    .orElse(null);
```

Updated.